src/compiler/sys.ts

```diff
@@ -500,7 +500,10 @@ namespace ts {
                     for (const current of files) {
                         const name = combinePaths(path, current);
                         if (!contains(exclude, getCanonicalPath(name))) {
-                            const stat = _fs.statSync(name);
+                            // fs.statSync would throw an exception if the file is a symlink
+                            // whose linked file doesn't exist. fs.lstatSync would return a stat
+                            // object for the symlink file itself in this case
+                            const stat = _fs.lstatSync(name);

```

Comment:
This seems like a non-trivial change with a lot of nuance (for example, this means we won't traverse into directories now if they are created as symbolic links, as `isDirectory()` returns `false` on a `lstat` where it used to return `true` on a `stat`.

How is this part of setting a limit on file size when loading? Seems like this is a different issue and should be coded/evaluated separately.